asserts/asserts.go

```diff
@@ -1399,3 +1407,86 @@ func SignatureCheck(assert Assertion, pubKey PublicKey) error {
 	}
 	return nil
 }
+
+// VerifySignature checks the signature of a given boady against the given public key.
+// It is used by model service to verify nonce of the request-id.
+// It currently supports key with algorithms RSA, DSA, ECDSA, and ED25519.
+// All data is expected to use the SHA3-384 hash.
+func VerifySignature(nonce, signature []byte, pubKey crypto.PublicKey) error {
```

Comment:
my preference would be for this to have a more specific name and take a HardwareIdentiy assertion directly?  then it can live in the file for it, and not here

I also noticed that HardwareIdentity itself doesn't have a doc comment, can you add one here or a separate PR